django/db/backends/oracle/schema.py

```diff
@@ -56,22 +63,34 @@ def alter_field(self, model, old_field, new_field, strict=False):
             # SQLite-ish workaround
             if 'ORA-22858' in description or 'ORA-22859' in description:
                 self._alter_field_type_workaround(model, old_field, new_field)
+            # If an identity column is changing to a non-numeric type, drop the
+            # identity first.
+            elif 'ORA-30675' in description:
+                self._drop_identity(model._meta.db_table, old_field.column)
+                self.alter_field(model, old_field, new_field, strict)
+            # If a primary key column is changing to an identity column, drop
+            # the primary key first.
+            elif 'ORA-30673' in description and old_field.primary_key:
+                self._delete_primary_key(model, strict=True)
+                self._alter_field_type_workaround(model, old_field, new_field)
             else:
                 raise
 
     def _alter_field_type_workaround(self, model, old_field, new_field):
         """
         Oracle refuses to change from some type to other type.
         What we need to do instead is:
-        - Add a nullable version of the desired field with a temporary name
+        - Add a nullable version of the desired field with a temporary name. If
+          the new column is an auto field, then the temporary column can't be
+          nullable.
         - Update the table to transfer values from old to new
         - Drop old column
         - Rename the new column and possibly drop the nullable property
         """
         # Make a new field that's like the new one but with a temporary
         # column name.
         new_temp_field = copy.deepcopy(new_field)
-        new_temp_field.null = True
+        new_temp_field.null = (new_field.get_internal_type() != 'AutoField')
```

Comment:
Good catch! Thanks, I fixed this and added a new test: *schema.tests.SchemaTests.test_alter_int_pk_to_bigautofield_pk*.